CMakeLists.txt

```diff
@@ -157,66 +123,102 @@ endif()
-set_property(CACHE KEEPASSXC_DIST_TYPE PROPERTY STRINGS Snap AppImage Flatpak Other)
+set(KEEPASSXC_DIST_TYPE "Native" CACHE STRING "KeePassXC Distribution Type")
+set_property(CACHE KEEPASSXC_DIST_TYPE PROPERTY STRINGS Snap AppImage Flatpak Native)
 if(KEEPASSXC_DIST_TYPE STREQUAL "Snap")
     set(KEEPASSXC_DIST_SNAP ON)
 elseif(KEEPASSXC_DIST_TYPE STREQUAL "AppImage")
     set(KEEPASSXC_DIST_APPIMAGE ON)
 elseif(KEEPASSXC_DIST_TYPE STREQUAL "Flatpak")
     set(KEEPASSXC_DIST_FLATPAK ON)
-elseif(KEEPASSXC_DIST_TYPE STREQUAL "Other")
-    unset(KEEPASSXC_DIST)
 endif()
 
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.14.0")
-    cmake_policy(SET CMP0083 NEW)
-    include(CheckPIESupported)
-    check_pie_supported()
+# Standards
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Compiler Features
+if(APPLE)
+    # Perform the platform checks before applying the stricter compiler flags.
+    # Otherwise the kSecAccessControlTouchIDCurrentSet deprecation warning will result in an error.
+    try_compile(XC_APPLE_COMPILER_SUPPORT_BIOMETRY
+       ${CMAKE_BINARY_DIR}/macos-trycompile/
+       ${CMAKE_SOURCE_DIR}/cmake/compiler-checks/control_biometry_support.mm)
+    message(STATUS "Biometry compiler support: ${XC_APPLE_COMPILER_SUPPORT_BIOMETRY}")
+
+    try_compile(XC_APPLE_COMPILER_SUPPORT_TOUCH_ID
+       ${CMAKE_BINARY_DIR}/macos-trycompile/
+       ${CMAKE_SOURCE_DIR}/cmake/compiler-checks/control_touch_id_support.mm)
+    message(STATUS "Touch ID compiler support: ${XC_APPLE_COMPILER_SUPPORT_TOUCH_ID}")
+
+    try_compile(XC_APPLE_COMPILER_SUPPORT_WATCH
+       ${CMAKE_BINARY_DIR}/macos-trycompile/
+       ${CMAKE_SOURCE_DIR}/cmake/compiler-checks/control_watch_support.mm)
+    message(STATUS "Apple watch compiler support: ${XC_APPLE_COMPILER_SUPPORT_WATCH}")
+
+    try_compile(HAVE_PT_DENY_ATTACH
+	   ${CMAKE_BINARY_DIR}/macos-trycompile/
+	   ${CMAKE_SOURCE_DIR}/cmake/compiler-checks/ptrace_deny_attach.cpp)
 endif()
 
-# Find Botan early since the version affects subsequent compiler options
-find_package(Botan REQUIRED)
-if(BOTAN_VERSION VERSION_GREATER_EQUAL "3.0.0")
-    set(WITH_XC_BOTAN3 TRUE)
-elseif(BOTAN_VERSION VERSION_LESS "2.11.0")
-    # Check for minimum Botan version
-    message(FATAL_ERROR "Botan 2.11.0 or higher is required")   
+if(UNIX)
+    check_cxx_source_compiles("#include <sys/prctl.h>
+    int main() { prctl(PR_SET_DUMPABLE, 0); return 0; }"
+            HAVE_PR_SET_DUMPABLE)
+
+    check_cxx_source_compiles("#include <malloc.h>
+    int main() { return 0; }"
+            HAVE_MALLOC_H)
+
+    check_cxx_source_compiles("#include <malloc.h>
+    int main() { malloc_usable_size(NULL); return 0; }"
+            HAVE_MALLOC_USABLE_SIZE)
+
+    check_cxx_source_compiles("#include <sys/resource.h>
+    int main() {
+      struct rlimit limit;
+      limit.rlim_cur = 0;
+      limit.rlim_max = 0;
+      setrlimit(RLIMIT_CORE, &limit);
+      return 0;
+    }" HAVE_RLIMIT_CORE)
```

Comment:
Should these snippets be moved into individual files in `cmake/compiler-checks/` for consistency?

Reply:
I considered it, but that requires a much higher cmake version to avoid the boilerplate code. I believe it is 3.20 which isn't available on most Linux distros